libs/api/domains/financial-aid/src/lib/financialAid.module.ts

```diff
@@ -0,0 +1,36 @@
+import { Module, DynamicModule } from '@nestjs/common'
+import { FinancialAidResolver } from './financialAid.resolver'
+import { FinancialAidService } from './financialAid.service'
+import { ApplicationApi, Configuration } from '../../gen/fetch'
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+
+export interface Config {
+  baseApiUrl: string
+}
+
+@Module({})
+export class FinancialAidModule {
+  static register(config: Config): DynamicModule {
+    return {
+      module: FinancialAidModule,
+      providers: [
+        FinancialAidResolver,
+        FinancialAidService,
+        {
+          provide: ApplicationApi,
```

Comment:
Should move this to a stand-alone client library so you can share it with your `submitApplication` template API.

Reply:
Great point, then I can also use that client in our api and remove rest-datasource as the way to talk to our backend from our api, will do 🙌

Reply:
Ooh, that would be nice!